### PR TITLE
Remove VPA `(min|max)Allowed` settings.

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area networking
/area auto-scaling
/kind enhancement

**What this PR does / why we need it**:

Remove VPA `(min|max)Allowed` settings.

`minAllowed` may cause wasted resources while `maxAllowed` may cause under-reservation and hence noisy neighbor issues. This change removes them building on previous work in #489. They are removed from `kube-controllers`, `calico-typha`, the admission controller and the extension.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The networking calico extension no longer configures min/maxAllowed in any managed VPA resource.
```

### CODE REVIEW

File: charts/gardener-extension-admission-calico/values.yaml [hunk 1]
```diff
@@ -13,12 +13,12 @@ global:
...
-        memory: 64Mi
-      maxAllowed:
-        cpu: 4
-        memory: 10G
+    resourcePolicy: {}
+#     minAllowed:
+#       memory: 64Mi
+#     maxAllowed:
+#       cpu: 4
+#       memory: 10G
```
Comment (axel7born) [inline charts/gardener-extension-admission-calico/values.yaml:21]: Why is this commented and not removed?

**Review by DockToFuture — APPROVED** [review]
/lgtm

File: charts/gardener-extension-admission-calico/values.yaml (hunk 1 above)
Comment (ScheererJ) [reply charts/gardener-extension-admission-calico/values.yaml:21]: It is still possible to set it by specifying it. The comments make it easy to see how to set it when deploying the extension/admission controller.

**Review by axel7born — APPROVED** [review]

**Pull request merged**